source/constructs/api/data_source/service.py

```diff
@@ -361,13 +381,338 @@
     if credentials is None:
         raise BizException(MessageEnum.SOURCE_ASSUME_ROLE_FAILED.get_code(),
                            MessageEnum.SOURCE_ASSUME_ROLE_FAILED.get_msg())
+    security_groups = ec2_client.describe_security_groups(GroupNames=[const.SECURITY_GROUP_JDBC])["SecurityGroups"]
+    if not security_groups:
+        raise BizException(MessageEnum.SOURCE_SECURITY_GROUP_NOT_EXISTS.get_code(),
+                           MessageEnum.SOURCE_SECURITY_GROUP_NOT_EXISTS.get_msg())
+    security_group = filter(lambda sg: sg["GroupName"] == const.SECURITY_GROUP_JDBC, security_groups)[0]
+    inbound_route = security_group["IpPermissions"][0]
+    if inbound_route["IpProtocol"] != "tcp" or not inbound_route["FromPort"] != 0 or not inbound_route["ToPort"] != 65535 \
+       or not inbound_route["IpRanges"] or inbound_route["IpRanges"][0]["CidrIp"] != "0.0.0.0/0":
+        raise BizException(MessageEnum.SOURCE_SG_INBOUND_ROUTE_NOT_VALID.get_code(),
+                           MessageEnum.SOURCE_SG_INBOUND_ROUTE_NOT_VALID.get_msg())
+    outbound_route = security_group["IpPermissionsEgress"][0]
+    if outbound_route["IpProtocol"] != "-1" or not outbound_route["IpRanges"] or outbound_route["IpRanges"][0]["CidrIp"] != "0.0.0.0/0":
+        raise BizException(MessageEnum.SOURCE_SG_OUTBOUND_ROUTE_NOT_VALID.get_code(),
+                           MessageEnum.SOURCE_SG_OUTBOUND_ROUTE_NOT_VALID.get_msg())
+    subnet = ec2_client.describe_subnets(SubnetIds=[jdbc.network_subnet_id])["Subnets"]
+    if not subnet:
+        raise BizException(MessageEnum.SOURCE_SUBNET_JDBC_NOT_EXISTS.get_code(),
+                           MessageEnum.SOURCE_SUBNET_JDBC_NOT_EXISTS.get_msg())
+    if subnet[0]["MapPublicIpOnLaunch"]:
+        raise BizException(MessageEnum.SOURCE_SUBNET_NOT_PRIVATE.get_code(),
+                           MessageEnum.SOURCE_SUBNET_NOT_PRIVATE.get_msg())
+    if not ec2_client.describe_nat_gateways(Filters=[{'Name': 'subnet-id', 'Values': [jdbc.network_subnet_id]}])['NatGateways']:
+        raise BizException(MessageEnum.SOURCE_SUBNET_NOT_CONTAIN_NAT.get_code(),
+                           MessageEnum.SOURCE_SUBNET_NOT_CONTAIN_NAT.get_msg())
+    if not ec2_client.describe_availability_zones(Filters=[{'Name': 'zone-name', 'Values': [jdbc.network_availability_zone]}])["AvailabilityZones"]:
+        raise BizException(MessageEnum.SOURCE_AVAILABILITY_ZONE_NOT_EXISTS.get_code(),
+                           MessageEnum.SOURCE_AVAILABILITY_ZONE_NOT_EXISTS.get_msg())
+
+def sync(jdbc: SourceJDBCConnection):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This function contains 87 lines of code, not including blank lines or lines with only comments, Python punctuation characters, identifiers, or literals. By comparison, 99% of the functions in the CodeGuru reference dataset contain fewer lines of code. Large functions might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this function or break it into multiple functions.